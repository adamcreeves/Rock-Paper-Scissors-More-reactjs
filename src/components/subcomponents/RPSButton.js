import React from "react";
import "../../styles/Main.css";

function RPSButton({ buttonFunc, label }) {
  return (
    <button className="button" onClick={buttonFunc}>
      {label}
    </button>
  );
}

export default RPSButton;
