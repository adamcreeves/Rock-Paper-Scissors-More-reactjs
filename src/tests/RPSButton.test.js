import React from "react";
import { render } from "@testing-library/react";
import RPSButton from "../components/subcomponents/RPSButton";

describe("Test RPSButton", () => {
  it("should match the snapshot", () => {
    const { asFragment } = render(<RPSButton />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("should render RPSButton and not be null", () => {
    const { asFragment } = render(<RPSButton />);
    expect(asFragment()).not.toBeNull();
  });
});
