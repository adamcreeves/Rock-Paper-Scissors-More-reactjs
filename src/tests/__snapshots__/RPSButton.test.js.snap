// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RPSButton should match the snapshot 1`] = `
<DocumentFragment>
  <button />
</DocumentFragment>
`;
